src/store/createStore.ts

```diff
@@ -1,8 +1,9 @@
 import Store from "../interfaces/Store"
 
-export default function createStore(state = {}): Store {
+export default function createStore(state = {}, middleware = null): Store {
```

Comment:
Why `middleware = null` instead of `middleware = []` ?

Reply:
@matheusml middleware is expected to be a function. not an array.

The return signature for applyMiddleware is as follows:

```ts
applyMiddleware(...middleware: Array<Function>): Function
```

Reply:
Yes, a middleware is a function, but I'm guessing you could pass multiple middlewares to `createStore`. Am I wrong?

Reply:
yes, it supports multiple middlewares BUT does it as part of the `applyMiddleware` function. 

`applyMiddleware` takes the role of handling and executing all the middlewares.

so its  

```js
createStore({}, applyMiddleware(middleware1, middleware2))
```